src/compiler/parser.ts

```diff
@@ -7305,7 +7312,13 @@ namespace ts {
             export function parseJSDocNameReference(): JSDocNameReference {
                 const pos = getNodePos();
                 const hasBrace = parseOptional(SyntaxKind.OpenBraceToken);
-                const entityName = parseEntityName(/* allowReservedWords*/ false);
+                const p2 = getNodePos();
+                let entityName: EntityName | JSDocInstanceReference = parseEntityName(/* allowReservedWords*/ false);
+                while (token() === SyntaxKind.PrivateIdentifier) {

```

Comment:
`while`?

So are you supporting `Foo#Bar#Baz`?

Reply:
A left-deep JSDocInstanceReference tree: `((Foo#Bar)#Baz)`